src/test/regress/sql/multi_foreign_key_relation_graph.sql

```diff
@@ -307,6 +307,169 @@ SELECT oid::regclass::text AS tablename
+ALTER TABLE local_table_1 ADD COLUMN unrelated_column int;
+
+-- show that droping a table not referenced and not referencing to any table
+-- does not invalidate foreign key graph
+DROP TABLE local_table_4;
+
+CREATE TABLE partitioned_table_1 (col_1 INT UNIQUE, col_2 INT) PARTITION BY RANGE (col_1);
+CREATE TABLE partitioned_table_1_100_200 PARTITION OF partitioned_table_1 FOR VALUES FROM (100) TO (200);
+CREATE TABLE partitioned_table_1_200_300 PARTITION OF partitioned_table_1 FOR VALUES FROM (200) TO (300);
+SELECT create_distributed_table('partitioned_table_1', 'col_1');
+
+CREATE TABLE partitioned_table_2 (col_1 INT UNIQUE, col_2 INT) PARTITION BY RANGE (col_1);
+CREATE TABLE partitioned_table_2_100_200 PARTITION OF partitioned_table_2 FOR VALUES FROM (100) TO (200);
+CREATE TABLE partitioned_table_2_200_300 PARTITION OF partitioned_table_2 FOR VALUES FROM (200) TO (300);
+SELECT create_distributed_table('partitioned_table_2', 'col_1');
+
+CREATE TABLE reference_table_4 (col_1 INT UNIQUE, col_2 INT UNIQUE);
+SELECT create_reference_table('reference_table_4');
+
+-- observe foreign key graph invalidation with partitioned tables
+ALTER TABLE partitioned_table_1 ADD CONSTRAINT fkey_8 FOREIGN KEY (col_1) REFERENCES reference_table_4(col_2);
+ALTER TABLE partitioned_table_2 ADD CONSTRAINT fkey_9 FOREIGN KEY (col_1) REFERENCES reference_table_4(col_2);
+
+CREATE TABLE partitioned_table_1_300_400 PARTITION OF partitioned_table_1 FOR VALUES FROM (300) TO (400);
+
+-- we invalidate foreign key graph as attach partition creates a new distributed table
+SELECT oid::regclass::text AS tablename
+FROM get_foreign_key_connected_relations('partitioned_table_1_300_400') AS f(oid oid)
+ORDER BY tablename;
+
+CREATE TABLE local_partitioned_table_1 (col_1 INT UNIQUE, col_2 INT) PARTITION BY RANGE (col_1);
+CREATE TABLE local_table_5 (col_1 INT UNIQUE, col_2 INT);
+
+-- in below two show that attaching a partition doesn't invalidate
+-- foreign key cache as parent table isn't involved in any foreign
+-- key relationship
+
+CREATE TABLE local_partitioned_table_1_100_200 PARTITION OF local_partitioned_table_1 FOR VALUES FROM (100) TO (200);
+
+CREATE TABLE local_partitioned_table_1_200_300 (col_1 INT UNIQUE, col_2 INT);
+ALTER TABLE local_partitioned_table_1 ATTACH PARTITION local_partitioned_table_1_200_300 FOR VALUES FROM (200) TO (300);
+
+-- define a foreign key from partitioned table
+ALTER TABLE local_partitioned_table_1 ADD CONSTRAINT fkey_10 FOREIGN KEY (col_1) REFERENCES local_table_5(col_1);
+
+-- in below two show that attaching partition invalidates foreign
+-- key cache as parent table is involved in a foreign key relationship
+
+CREATE TABLE local_partitioned_table_1_300_400 PARTITION OF local_partitioned_table_1 FOR VALUES FROM (300) TO (400);
+
+CREATE TABLE local_partitioned_table_1_500_600 (col_1 INT UNIQUE, col_2 INT);
+ALTER TABLE local_partitioned_table_1 ATTACH PARTITION local_partitioned_table_1_500_600 FOR VALUES FROM (500) TO (600);
```

Comment:
Lets add `get_foreign_key_connected_relations ` calls here. Also, I cannot see the tests that I commented as pointing the bug?